libsolidity/interface/CompilerStack.cpp

```diff
@@ -377,25 +374,26 @@ bool CompilerStack::analyze()
 			m_unhandledSMTLib2Queries += modelChecker.unhandledQueries();
 		}
 	}
-	catch(FatalError const&)
+	catch (FatalError const&)
 	{
 		if (m_errorReporter.errors().empty())
 			throw; // Something is weird here, rather throw again.
 		noErrors = false;
 	}
 
-	if (noErrors)
-	{
-		m_stackState = AnalysisSuccessful;
-		return true;
-	}
-	else
-		return false;
+	m_stackState = AnalysisPerformed;
+	if (!m_hasError && !noErrors)
```

Comment:
```suggestion
	if (!noErrors)
```

Same effect, but easier to read IMO.